config/azure.go

```diff
@@ -81,23 +80,18 @@ func (c *Config) azureEnsureWorkspaceUrl(ctx context.Context, ahr azureHostResol
 	}
 	// azure resource ID can also be used in lieu of host by some of the clients, like Terraform
 	management := ahr.tokenSourceFor(ctx, c, env, env.ResourceManagerEndpoint)
-	resourceManager := oauth2.NewClient(ctx, management)
-	resp, err := resourceManager.Get(env.ResourceManagerEndpoint + c.AzureResourceID + "?api-version=2018-04-01")
-	if err != nil {
-		return fmt.Errorf("cannot resolve workspace: %w", err)
-	}
-	raw, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return fmt.Errorf("cannot read: %w", err)
-	}
 	var workspaceMetadata struct {
 		Properties struct {
 			WorkspaceURL string `json:"workspaceUrl"`
 		} `json:"properties"`
 	}
-	err = json.Unmarshal(raw, &workspaceMetadata)
+	requestURL := env.ResourceManagerEndpoint + c.AzureResourceID + "?api-version=2018-04-01"
```

Comment:
It might be good to abstract out `"?api-version=2018-04-01"` and use a variable here instead since this could be used at other places in future.

Reply:
@tanmay-db not in this PR, unfortunately. there's `makeRequestBody`, that has the logic for query string & other things, but it has to be decoupled some day later.